s3/s3_transport/include/s3_transport.hpp

```diff
@@ -836,6 +835,29 @@ namespace irods::experimental::io::s3_transport
             return existing_object_size_;
         }
 
+        bool is_open_cache_file() {
+            if (use_cache_) {
```

Comment:
Same thing here - this could be flattened by reversing the logic:
```c++
if (!use_cache_) {
    return false;
}
// ...the rest of the stuff
```

Also, I think the name could use a tweak, unless I'm misunderstanding. Maybe one of these...
- is_cache_file_open
- has_open_cache_file
- cache_file_is_open

Just a semantics thing - the stream itself is not an "open cache file". Not a big deal.

Reply:
Flipped the logic and renamed the method to `is_cache_file_open()`.